lib/pure/parsexml.nim

```diff
@@ -446,13 +447,15 @@ proc parseTag(my: var XmlParser) =
     # an attribute follows:
     my.kind = xmlElementOpen
     my.state = stateAttr
-    my.c = my.a # save for later
+    # save for later:
+    my.c.setLen(my.a.len)
+    copyMem(addr my.c[0], addr my.a[0], my.a.len+1)
```

Comment:
I actually tried with swap and the resulting file ended up half as big, so something is going wrong. Since I don't understand what's happening here exactly I chose to do something with the same semantics.